source/addonStore/network.py

```diff
@@ -110,7 +111,13 @@ def download(
 		f.add_done_callback(self._done)
 
 	def _done(self, downloadAddonFuture: Future[Optional[os.PathLike]]):
-		isCancelled = downloadAddonFuture.cancelled() or downloadAddonFuture not in self._pending
+		isCancelled = (
+			downloadAddonFuture.cancelled()
+			or downloadAddonFuture not in self._pending
+			# When the add-on status is set to a value other than "DOWNLOADING" before the completeness check,
+			# the download is considered to be canceled externally.
+			or self._pending[downloadAddonFuture][0].status != AvailableAddonStatus.DOWNLOADING
```

Comment:
I don't think this is safe - shouldn't this be

```suggestion
			or self._pending[downloadAddonFuture][0].status is AvailableAddonStatus.DOWNLOAD_FAILED, 
```

Reply:
The reason why `! = AvailableAddonStatus.DOWNLOADING`, is that when the download is cancelled the status is changed by `listItemVM.status = getStatus(listItemVM.model, self._filteredStatusKey)` from `DOWNLOADING` to some other status.

Reply:
shouldn't in that case the following be true? why isn't the future cancelled and removed from pending when it is cancelled?
```
downloadAddonFuture.cancelled() or downloadAddonFuture not in self._pending
```

Reply:
You're right, even though I've added the code to ensure that the temporarily downloaded file is deleted, then I shouldn't be cleaning up the file with the help of the code here.